### PR TITLE
Predefine default field ordering by using `@orderBy` on fields

### PR DESCRIPTION
**Pull request by olivernybroe**
- [x] Added or updated tests
- [x] Documented user facing changes
- [x] Updated CHANGELOG.md

**Changes**
Add field support for order by directive.

This PR makes the order by directive works on fields also. 
This is handy for when you need to add an order by which is always applied to a field. For example so users always returns the newest user first.

**Breaking changes**

No BC's.

### CODE REVIEW

**Comment by olivernybroe** [comment]
@spawnia Should this be split into it's own directive file then? Or should change so the same directive can be defined multiple times in one directive class?

File: src/OrderBy/OrderByDirective.php [hunk 1]
```diff
@@ -39,6 +40,22 @@ public static function definition(): string
     """
     columnsEnum: String
 ) on ARGUMENT_DEFINITION
+
+"""
+Sort a result list by a column
+"""
+directive @orderBy(
```
Comment (spawnia) [inline src/OrderBy/OrderByDirective.php:47]: That will lead to errors in the generated `schema-directives.graphql`, directives names must be unique. You can just merge the two definitions and clarify correct usage in the description.

File: tests/Integration/OrderBy/OrderByDirectiveDBTest.php [hunk 2]
```diff
@@ -15,6 +15,8 @@ class OrderByDirectiveDBTest extends DBTestCase
             orderByRestricted: _ @orderBy(columns: ["name"])
             orderByRestrictedEnum: _ @orderBy(columnsEnum: "UserColumn")
         ): [User!]! @all
+
+        latestUsers: [User!]! @all @orderBy(column: "created_at" direction: "DESC")
```
Comment (spawnia) [inline tests/Integration/OrderBy/OrderByDirectiveDBTest.php:19]: I am trying to move away from having a single large schema per test class, it is rather bad for maintenance. Please define it down in the individual test method instead.

File: src/OrderBy/OrderByDirective.php [hunk 3]
```diff
@@ -84,4 +101,15 @@ public function manipulateArgDefinition(
             $argDefinition->type = Parser::typeReference('['.OrderByServiceProvider::DEFAULT_ORDER_BY_CLAUSE.'!]');
         }
     }
+
+    /**
+     * @param \Illuminate\Database\Query\Builder|\Illuminate\Database\Eloquent\Builder $builder
```
Comment (spawnia) [inline src/OrderBy/OrderByDirective.php:106]:
```suggestion
     * @param  \Illuminate\Database\Query\Builder|\Illuminate\Database\Eloquent\Builder  $builder
     * @return \Illuminate\Database\Query\Builder|\Illuminate\Database\Eloquent\Builder
```

**Review by spawnia — CHANGES_REQUESTED** [review]

File: src/OrderBy/OrderByDirective.php (hunk 1 above)
Comment (olivernybroe) [reply src/OrderBy/OrderByDirective.php:47]: Oh, I thought they just had to be unique based on where they were applied.

**Review by spawnia — APPROVED** [review]
@olivernybroe bouncing this back to you, feel free to merge if you agree with my changes.

**Comment by olivernybroe** [comment]
@spawnia I agree with the changes! :) 

Don't have permission to merge unfortunately.

**Pull request merged**